# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wgpu/wgpu-core/src/device/trace.rs
# Commit: ef5dc3e04e5f
# Full Hash: ef5dc3e04e5f271eea0636ab3a495e95cc912f1d
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2021-09-04 09:40:24
# Regressor Bug: 1726626
# File Overlap Count: 3
# Description:
#   Bug 1726626 - Move gfx/wgpu into a 3rd party dependency r=jgilbert,bholley
#   
#   This update makes wgpu a vendored dependency instead of having it in gfx/wgpu.
#   
#   ## Notes
# ==============================================================================

diff --git a/gfx/wgpu/wgpu-core/src/device/trace.rs b/third_party/rust/wgpu-core/src/device/trace.rs
rename from gfx/wgpu/wgpu-core/src/device/trace.rs
rename to third_party/rust/wgpu-core/src/device/trace.rs
--- a/gfx/wgpu/wgpu-core/src/device/trace.rs
+++ b/third_party/rust/wgpu-core/src/device/trace.rs
@@ -1,65 +1,69 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
 use crate::id;
 use std::ops::Range;
 #[cfg(feature = "trace")]
 use std::{borrow::Cow, io::Write as _};
 
 //TODO: consider a readable Id that doesn't include the backend
 
 type FileName = String;
 
 pub const FILE_NAME: &str = "trace.ron";
 
 #[cfg(feature = "trace")]
 pub(crate) fn new_render_bundle_encoder_descriptor<'a>(
     label: crate::Label<'a>,
     context: &'a super::RenderPassContext,
+    is_ds_read_only: bool,
 ) -> crate::command::RenderBundleEncoderDescriptor<'a> {
     crate::command::RenderBundleEncoderDescriptor {
         label,
         color_formats: Cow::Borrowed(&context.attachments.colors),
-        depth_stencil_format: context.attachments.depth_stencil,
-        sample_count: context.sample_count as u32,
+        depth_stencil: context.attachments.depth_stencil.map(|format| {
+            let aspects = hal::FormatAspects::from(format);
+            wgt::RenderBundleDepthStencil {
+                format,
+                depth_read_only: is_ds_read_only && aspects.contains(hal::FormatAspects::DEPTH),
+                stencil_read_only: is_ds_read_only && aspects.contains(hal::FormatAspects::STENCIL),
+            }
+        }),
+        sample_count: context.sample_count,
     }
 }
 
 #[allow(clippy::large_enum_variant)]
 #[derive(Debug)]
 #[cfg_attr(feature = "trace", derive(serde::Serialize))]
 #[cfg_attr(feature = "replay", derive(serde::Deserialize))]
 pub enum Action<'a> {
     Init {
         desc: crate::device::DeviceDescriptor<'a>,
         backend: wgt::Backend,
     },
+    ConfigureSurface(id::SurfaceId, wgt::SurfaceConfiguration),
     CreateBuffer(id::BufferId, crate::resource::BufferDescriptor<'a>),
     FreeBuffer(id::BufferId),
     DestroyBuffer(id::BufferId),
     CreateTexture(id::TextureId, crate::resource::TextureDescriptor<'a>),
     FreeTexture(id::TextureId),
     DestroyTexture(id::TextureId),
     CreateTextureView {
         id: id::TextureViewId,
         parent_id: id::TextureId,
         desc: crate::resource::TextureViewDescriptor<'a>,
     },
     DestroyTextureView(id::TextureViewId),
     CreateSampler(id::SamplerId, crate::resource::SamplerDescriptor<'a>),
     DestroySampler(id::SamplerId),
-    CreateSwapChain(id::SwapChainId, wgt::SwapChainDescriptor),
-    GetSwapChainTexture {
-        id: id::TextureViewId,
-        parent_id: id::SwapChainId,
+    GetSurfaceTexture {
+        id: id::TextureId,
+        parent_id: id::SurfaceId,
     },
-    PresentSwapChain(id::SwapChainId),
+    Present(id::SurfaceId),
     CreateBindGroupLayout(
         id::BindGroupLayoutId,
         crate::binding_model::BindGroupLayoutDescriptor<'a>,
     ),
     DestroyBindGroupLayout(id::BindGroupLayoutId),
     CreatePipelineLayout(
         id::PipelineLayoutId,
         crate::binding_model::PipelineLayoutDescriptor<'a>,
@@ -93,17 +97,17 @@ pub enum Action<'a> {
     CreateRenderBundle {
         id: id::RenderBundleId,
         desc: crate::command::RenderBundleEncoderDescriptor<'a>,
         base: crate::command::BasePass<crate::command::RenderCommand>,
     },
     DestroyRenderBundle(id::RenderBundleId),
     CreateQuerySet {
         id: id::QuerySetId,
-        desc: wgt::QuerySetDescriptor,
+        desc: crate::resource::QuerySetDescriptor<'a>,
     },
     DestroyQuerySet(id::QuerySetId),
     WriteBuffer {
         id: id::BufferId,
         data: FileName,
         range: Range<wgt::BufferAddress>,
         queued: bool,
     },